File: modules/coupon-domain/src/main/java/com/thanh0x/coursedeal/dto/CouponRequestDTO.java
package com.thanh0x.coursedeal.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Request body for creating a new coupon.
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class CouponRequestDTO {
    /**
     * The Udemy coupon URL.
     */
    @NotBlank(message = "Coupon URL is required")
    private String couponUrl;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/CouponRequestDTO.kt
package com.thanh0x.coursedeal.dto

import jakarta.validation.constraints.NotBlank

/**
 * Request body for creating a new coupon.
 */
data class CouponRequestDTO(
    /**
     * The Udemy coupon URL.
     */
    @field:NotBlank(message = "Coupon URL is required")
    val couponUrl: String = ""
)
